geowebcache/core/src/main/java/org/geowebcache/config/XMLFileResourceProvider.java

```diff
@@ -0,0 +1,277 @@
+    }
+
+    @Override
+    public void backup() throws IOException {
+        backUpConfig(findOrCreateConfFile());
+    }
+
+    @Override
+    public String getId() {
+        return configDirectory.getAbsolutePath();
+    }
+
+    private static String getConfigDirVar(ApplicationContext ctxt){
+        return GWCVars.findEnvVar(ctxt, GWC_CONFIG_DIR_VAR);
+    }
+    
+
+    @Override
+    public void setTemplate(final String templateLocation) {
+        this.templateLocation = templateLocation;
+    }
+
+    private File findConfigFile() throws IOException {
+        if (null == configDirectory) {
+            throw new IllegalStateException();
+        }
+
+        if (!configDirectory.exists() && !configDirectory.mkdirs()) {
+            throw new IOException(
+                    "Configuration directory does not exist and cannot be created: '"
+                            + configDirectory.getAbsolutePath() + "'");
+        }
+        if (!configDirectory.canWrite()) {
+            throw new IOException("Configuration directory is not writable: '"
+                    + configDirectory.getAbsolutePath() + "'");
+        }
+
+        File xmlFile = new File(configDirectory, configFileName);
+        return xmlFile;
+    }
+    
+    public String getLocation() throws IOException {
+        File f = findConfigFile();
+        try {
+            return f.getCanonicalPath();
+        } catch (IOException ex) {
+            log.error("Could not canonize config path", ex);
+            return f.getPath();
+        }
+    }
+
+    private File findOrCreateConfFile() throws IOException {
+        File xmlFile = findConfigFile();
+
+        if (xmlFile.exists()) {
+            log.info("Found configuration file in " + configDirectory.getAbsolutePath());
+        } else if (templateLocation != null) {
+            log.warn("Found no configuration file in config directory, will create one at '"
+                    + xmlFile.getAbsolutePath() + "' from template "
+                    + getClass().getResource(templateLocation).toExternalForm());
+            // grab template from classpath
+            try {
+                InputStream templateStream = getClass().getResourceAsStream(templateLocation);
+                try {
+                    OutputStream output = new FileOutputStream(xmlFile);
+                    try {
+                        IOUtils.copy(templateStream, output);
+                    } finally {
+                        output.flush();
+                        output.close();
+                    }
+                } finally {
+                    templateStream.close();
+                }
+            } catch (IOException e) {
+                throw new IOException("Error copying template config to "
+                        + xmlFile.getAbsolutePath(), e);
+            }
+
+        }
+
+        return xmlFile;
+    }
+    
+
+    private void backUpConfig(final File xmlFile) throws IOException {
```

Comment:
@NielsCharlier as this just been moved from another class? Can this code be shared with the very similar implementation in GeoServer, which appears to be mostly a copy of the code from GWC?